### PR TITLE
Resource leakage fixes

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

**Review by rallytime — APPROVED** [review]

File: salt/modules/inspectlib/kiwiproc.py [hunk 1]
```diff
@@ -145,19 +145,21 @@ def _set_users(self, node):
...
+                        shadow[login] = {'p': pwd}
+
+        with salt.utils.fopen('/etc/passwd') as rfh:
+            for ps_line in rfh.read().split(os.linesep):
+                if ps_line.strip():
+                    ps_line = ps_line.strip().split(':')
+                    if ps_line[0] in shadow:
+                        shadow[ps_line[0]]['h'] = ps_line[5]
+                        shadow[ps_line[0]]['s'] = ps_line[6]
+                        shadow[ps_line[0]]['g'] = self._get_user_groups(ps_line[0])
```
Comment (isbm) [inline salt/modules/inspectlib/kiwiproc.py]: Better. Thanks!

Comment (s0undt3ch) [inline salt/modules/inspectlib/kiwiproc.py]: We have to use the `with` context in order to assure the file is closed. If you turn off the `ResourceWarning` filter on the `warnings` module, py3 will complain.

Comment (isbm) [inline salt/modules/inspectlib/kiwiproc.py]: I am not saying anything. Just :+1: :+1: :+1: that. :laughing:

Comment (s0undt3ch) [inline salt/modules/inspectlib/kiwiproc.py]: Yeah, I know, just explaining why if anyone passing by doesn't know :)

**Pull request merged**